src/cockpit/channels/metrics.py

```diff
@@ -120,35 +120,42 @@ def sample(self):
             sampler.sample(samples)
         return samples
 
-    def calculate_sample_rate(self, value: Any, old_value: Optional[Any]):
+    def calculate_sample_rate(self, value: float, old_value: Optional[float]) -> Union[float, bool]:
         if old_value is not None and self.last_timestamp:
             return (value - old_value) / (self.next_timestamp - self.last_timestamp)
         else:
             return False
 
-    def send_updates(self, samples: Dict[str, Any], last_samples: Dict[str, Any]):
-        data = []
+    def send_updates(self, samples: Samples, last_samples: Samples):
+        data: List[Union[float, List[Optional[Union[float, bool]]]]] = []
```

Comment:
I think `Optional[Union[float, Literal[False]]` probably reads better as `Union[float, Literal[False], None]`, but that's taste.  Would you say "no change" exists at a semantically higher level than "failed to calculate"?  Maybe...

These types are getting big.  I wonder if it makes sense to define some symbolic names...